src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/ContigAlignmentsModifier.java

```diff
@@ -237,6 +237,120 @@ public String toString() {
         }
     }
 
+    private static class GapSplitter {
+        private static final int NOT_SET = -1;
+
+        int alignmentStartContig = NOT_SET;
+        int alignmentStartIdx = NOT_SET;
+        int alignmentStartRef = NOT_SET;
+        int alignmentEndContig = NOT_SET;
+        int alignmentEndIdx = NOT_SET;
+        int alignmentEndRef = NOT_SET;
+
+        final AlignmentInterval oneRegion;
+        final int unclippedContigLen;
+
+        public GapSplitter( final AlignmentInterval oneRegion, final int unclippedContigLen ) {
+            this.oneRegion = oneRegion;
+            this.unclippedContigLen = unclippedContigLen;
+        }
+
+        public List<AlignmentInterval> splitGaps( final int sensitivity ) {
+            final List<CigarElement> elements = oneRegion.cigarAlong5to3DirectionOfContig.getCigarElements();
```

Comment:
No, this is completely rewritten to circumvent the exception that was being thrown on cigars with adjacent insertion and deletion operators.  In the new code, leading and trailing indels are never included in the cigar pieces because the starting and ending positions are updated only on aligned bits (M operators), never on indels.